server/cmd/mattermost/main.go

```diff
@@ -17,6 +19,12 @@ import (
 )
 
 func main() {
+
+	// Check if the command is called by root user
+	if os.Geteuid() == 0 {
+		mlog.Warn("Running Mattermost as root is not recommended. Please use a separate user")
+	}
```

Comment:
While the code works correctly, I think it should be placed in `root.go` instead to keep `main.go` clean. You can use the `PersistentPreRun` field of `RootCmd`.